snuba/consumers/consumer_builder.py

```diff
@@ -217,10 +259,14 @@ def __build_consumer(
                 commit_retry_policy=self.__commit_retry_policy,
             )
 
-        return StreamProcessor(consumer, self.raw_topic, strategy_factory, IMMEDIATE)
+        raw_topic_spec = KafkaTopicSpec(SnubaTopic(self.raw_topic.name))
```

Comment:
This is not the correct usage of KafkaTopicSpec. You should never be constructing it yourself but just fetching it from the stream loader. ie. `stream_loader.get_default_topic_spec()` then calling `get_physical_topic_name` on that to resolve it.

Reply:
Ah ok. So if I follow correctly, what you mean is that I can support topic override as it is in un-sliced contexts already, and then get the physical topic name in this way?

Reply:
Yes. The KafkaTopicSpec is now taking care of all the sliced vs unsliced differences. So the consumer builder doesn't have to care about it anymore - as long as it passes the slice_id along to get_physical_topic_name(). Since that's taken care of in KafkaTopicSpec, this file should hardly change at all now. All you need to do is make sure that anywhere that was previously calling `KafkaTopicSpec.topic_name` now calls `KafkaTopicSpec.get_physical_topic_name(slice_id)` so that function can figure everything out taking the slice or lack of slice into account. There are 3 places that need to be replaced - default/raw topic, commit log topic and replacement topic - all these should be done the same way.

Reply:
Thanks, this makes sense and is a lot cleaner and more consistent!